examples/contrib/readhar.py

```diff
@@ -0,0 +1,141 @@
+        self.flows = []
+
+    def fix_headers(
+        self, request_headers: list[dict[str, str]] | list[tuple[str, str]]
+    ) -> list[tuple[bytes, bytes]]:
+        """Converts provided headers into (b"header-name", b"header-value") tuples"""
+        flow_headers = []
+        for header in request_headers:
+            # Applications that use the {"name":item,"value":item} notation are Brave, Chrome, Edge, Firefox, Charles, Fiddler, Insomnia, Safari
+            if isinstance(header, dict):
+                try:
+                    key = header["name"]
+                    value = header["value"]
+                except KeyError as e:
+                    raise exceptions.OptionsError(str(e)) from e
+
+            # Application that use the [name, value] notation is Slack
+            else:
+                try:
+                    key = header[0]
+                    value = header[1]
+                except IndexError as e:
+                    raise exceptions.OptionsError(str(e)) from e
+            flow_headers.append((bytes(key, "utf-8"), bytes(value, "utf-8")))
+
+        return flow_headers
+
+    # Don't know how to make a type annotation for the request json
+    def request_to_flow(self, request_json: dict) -> http.HTTPFlow:
+        """
+        Creates a HTTPFlow object from a given entry in HAR file
+        """
+
+        request_method = request_json["request"]["method"]
+        request_url = request_json["request"].get("url", None)
+        if not request_url:
+            request_url = request_json["request"].get("host", None)
+
+        request_headers = self.fix_headers(request_json["request"].get("headers", []))
+
+        client_conn = connection.Client(
+            peername=("127.0.0.1", 51513),
+            sockname=("127.0.0.1", 8080),
+            # TODO Get time info from HAR File
+            timestamp_start=time.time(),
+        )
+
+        # 375:3 is default mitmproxy server_conn when making a new flow.
+        server_conn = connection.Server(address=("375", 3))
+
+        newflow = http.HTTPFlow(client_conn, server_conn)
+        try:
+            newflow.request = http.Request.make(
+                request_method, request_url, "", request_headers
+            )
+        except TypeError as e:
+            logger.error("Failed to create request")
+            raise exceptions.OptionsError(str(e)) from e
+
+        response_code = request_json["response"].get("status", "")
+
+        response_content = request_json["response"]["content"].get("text", "")
+
+        response_headers = self.fix_headers(request_json["response"].get("headers", []))
+        try:
+            newflow.response = http.Response.make(
+                response_code, response_content, response_headers
+            )
+        except TypeError as e:
+            logger.error("Failed to create response")
+            raise exceptions.OptionsError(str(e)) from e
+
+        return newflow
+
+    async def load_flows(self) -> None:
```

Comment:
What happened to https://github.com/mitmproxy/mitmproxy/pull/6190#discussion_r1240470653? This seems overly complicated. I think we can avoid all instance state (including `self.flows`) for this addon.